src/ophys_etl/transforms/suite2p_wrapper.py

```diff
@@ -0,0 +1,219 @@
+                         "during triage, before refinement"))
+    high_pass = argschema.fields.Int(
+            default=100,
+            description=("Running mean subtraction with window of "
+                         "size 'high_pass'"))
+    smooth_masks = argschema.fields.Bool(
+            default=True,
+            description=("Whether to smooth masks in final pass of cell "
+                         "detection. Especially useful if one is in a high "
+                         "noise regime."))
+    # s2p ROI extraction parameters
+    inner_neuropil_radius = argschema.fields.Int(
+            default=2,
+            description=("Number of pixels to keep between ROI and neuropil "
+                         "donut."))
+    min_neuropil_pixels = argschema.fields.Int(
+            default=350,
+            description=("Minimum number of pixels in the neuropil."))
+    allow_overlap = argschema.fields.Bool(
+            # NOTE: changing this from extraction validation and labeling
+            # will be untangled downstream in demixing.
+            default=True,
+            description=("Pixels that are overlapping are thrown out (False) "
+                         "or added to both ROIs (True)"))
+    # Allen-specific options
+    bin_size = argschema.fields.Int(
+            required=False,
+            description=("If nbinned not provided, will calculate nbinned as "
+                         "nframes / bin_size"))
+    output_dir = argschema.fields.OutputDir(
+            required=True,
+            description="for minimal and cleaner output, specifies output dir")
+    retain_files = argschema.fields.List(
+            argschema.fields.Str,
+            cli_as_single_argument=True,
+            required=True,
+            default=['stat.npy'],
+            description=("only Suite2P output files with basenames in this "
+                         "list will be retained. If 'all', a complete list "
+                         "will be retained."))
+    timestamp = argschema.fields.Bool(
+            default=True,
+            description=("if true, will add a timestamp to the output file "
+                         "names"))
+
+    @mm.post_load
+    def check_args(self, data, **kwargs):
+        if ('nbinned' not in data) & ('bin_size' not in data):
+            raise Suite2PWrapperException(
+                    "must provide either `nbinned` or `bin_size`")
+        return data
+
+    @mm.post_load
+    def check_retain_files(self, data, **kwargs):
+        if 'all' in data['retain_files']:
+            data['retain_files'] = [
+                    'ops1.npy', 'data.bin', 'Fneu.npy', 'F.npy', 'iscell.npy',
+                    'ops.npy', 'spks.npy', 'stat.npy']
+        return data
+
+
+class Suite2PWrapperOutputSchema(argschema.schemas.DefaultSchema):
+    output_files = argschema.fields.Dict(
+        keys=argschema.fields.Str,
+        values=argschema.fields.OutputFile,
```

Comment:
Hmm, maybe this should actually be `InputFile`. I think for `OutputFile` argschema checks that one could write that file, not that it exists.

Reply:
changed this.
will consider an argschema change to have an alias for `InputFile` be `ReadableFile` and for `OutputFile` be `WriteableLocation`